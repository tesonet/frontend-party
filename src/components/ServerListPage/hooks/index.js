export { default as useFetchServers } from './useFetchServers';
export { default as useSortServers } from './useSortServers';
