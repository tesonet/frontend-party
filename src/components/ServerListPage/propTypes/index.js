import PropTypes from 'prop-types';
import {
  ASCENDING,
  DESCENDING,
  NAME,
  DISTANCE,
} from '../config/constants';

export const sortConfigDefaultProps = {
  [DISTANCE]: null,
  [NAME]: null,
};

export const sortConfigPropTypes = PropTypes.shape({
  [DISTANCE]: PropTypes.oneOf([ASCENDING, DESCENDING]),
  [NAME]: PropTypes.oneOf([ASCENDING, DESCENDING]),
});

export const serverPropTypes = PropTypes.shape({
  [NAME]: PropTypes.string,
  [DISTANCE]: PropTypes.number,
});
