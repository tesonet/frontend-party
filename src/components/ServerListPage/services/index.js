export { default as getServers } from './getServers';
