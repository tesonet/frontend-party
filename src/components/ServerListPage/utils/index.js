export { default as sortServers } from './sortServers';
