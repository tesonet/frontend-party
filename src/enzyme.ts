import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import fetchMock from "jest-fetch-mock";

declare let global: { fetch: {} };

global.fetch = fetchMock;
configure({ adapter: new Adapter() });
