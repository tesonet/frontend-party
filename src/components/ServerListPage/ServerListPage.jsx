import React from 'react';

import {
  withErrorWrapper,
  Loader,
  withErrorWrapperPropTypes,
} from '@Common';

import { useFetchServers, useSortServers } from './hooks';
import ServerList from './components/ServerList';
import { NO_SERVERS_MESSAGE } from './config/constants';

const ServerListPage = ({ errorHandler }) => {
  const {
    serversLoaded,
    servers,
    setServers,
  } = useFetchServers(errorHandler);
  const { sortConfig, handleSortAction } = useSortServers(servers, setServers);

  if (servers && servers.length === 0) {
    return NO_SERVERS_MESSAGE;
  }

  return (
    <Loader loaded={serversLoaded}>
      {servers && (
        <ServerList
          servers={servers}
          sortConfig={sortConfig}
          handleSortAction={handleSortAction}
        />
      )}
    </Loader>
  );
};

ServerListPage.propTypes = {
  ...withErrorWrapperPropTypes,
};

export default withErrorWrapper(ServerListPage);
