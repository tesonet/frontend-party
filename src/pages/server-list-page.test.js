import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";

import ServerListPage from "./server-list-page";


const component = shallow(<ServerListPage/>);

it('renders correctly', () => {
    expect(toJson(component)).toMatchSnapshot();
});
