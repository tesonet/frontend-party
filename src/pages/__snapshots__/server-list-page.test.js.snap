// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Fragment>
  <Connect(ServerListHeader) />
  <Connect(ServerListTable) />
</Fragment>
`;
